### ISSUE TITLE
benchmark: use output files instead of directories

### ISSUE THREAD

**Issue by tiborsimko** [reanahub/reana#560]
**Current behaviour**

When one launches many benchmark tests, the results are stored as follows:

```
benchmark_test200max80batch7job6clear/execution_progress.png
benchmark_test200max40batch7job6clear/execution_progress.png
...
```

and there could be hundreds of folders like that.

This has some inconveniences:

- using `-w foo` where `foo` is the desired identification string, one cannot immediately auto-complete in the shell using the same string out of the box, because the directory is prefixed with `benchmark_*`;
- sharing selected plots as results is not easy out of the box, because all files are called `execution_progress.png`, so one needs to rename them.

**Expected behaviour**

Using `-w foo` to launch tests, `foo` serves as the unique identifier for the given benchmark run, so we could use it everywhere. The benchmarking script could create the following:

```
test200max80batch7job6clear/test200max80batch7job6clear_original_results.csv
test200max80batch7job6clear/test200max80batch7job6clear_execution_progress.png
...
test200max40batch7job6clear/test200max40batch7job6clear_original_results.csv
test200max40batch7job6clear/test200max40batch7job6clear_histogram_pending_time.png
```

or simply in the current working directory:

```
test200max80batch7job6clear_original_results.csv
test200max80batch7job6clear_execution_progress.png
...
est200max40batch7job6clear_original_results.csv
test200max40batch7job6clear_execution_progress.png
```

which might be even better, as this is typically where new files are being created by Unix tools.  (And, if the user would like to use some directories, they could be easily moved later.)

So, basically, instead of `benchmark_<id>/<plot>`, we'd use `<id>_<plot>` naming strategy.

This should take care of the two issues described above.
